### PR TITLE
Update cost results presentation

### LINKED ISSUE #129: Bug: Only one bar in cost comparison plots

### PR DESCRIPTION
**Pull request by joAschauer**
Dataframes with costs for each source region, renewable energy generation, data scenario and supply chain are now calculated outside the tabs. Tab functions that need one of the cost dataframes receive these as arguments.

also closes #129

### CODE REVIEW

**Comment by joAschauer** [comment]
resolves # 131

File: ptxboa_streamlit.py [hunk 1]
```diff
@@ -90,29 +90,52 @@
...
+    parameter_list=[
+        x for x in api.get_dimension("res_gen").index.to_list() if x != "PV tracking"
+    ],
+    user_data=st.session_state["user_changes_df"],
+)
+costs_per_chain = pf.calculate_results_list(
+    api,
+    parameter_to_change="chain",
+    parameter_list=None,
+    user_data=st.session_state["user_changes_df"],
```
Comment (markushal) [inline ptxboa_streamlit.py:122]: why do you pass user_data as an argument? The function could access the session state directly

**Review by markushal — APPROVED** [review]
I like it! Definitely an improvement.

File: ptxboa_streamlit.py (hunk 1 above)
Comment (joAschauer) [reply ptxboa_streamlit.py]: changed accordingly

**Pull request merged**